internal/compiler/passes/focus_handling.rs

```diff
@@ -92,7 +92,7 @@ impl<'a> LocalFocusForwards<'a> {
 
             let Expression::ElementReference(focus_target) = &forward_focus_binding.expression
             else {
-                assert!(matches!(forward_focus_binding.expression, Expression::Invalid), "internal error: forward-focus property is of type ElementReference but received non-element-reference binding");
+                // resolve expressions pass will produce type error
```

Comment:
Hmm, I don't think that would hold. This is _before_ resolve expressions, so it would have to be `!diag.has_error()`, but that's also not in our hands as other code may have produced errors. So it really isn't "has produce", it's "will produce" :)

Reply:
Can't be. Resolving expression has to happen first. (otherwise you wouldn't see Expression::ElementReference)
(you see the errors after the panics because they are rpoerted at the end)

Reply:
Oh indeed. That explains it. Thanks :) I'll add your change.